lib/movie_widgets/movie_genres.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:movie_db_app/repositories_i/repository_i.dart';
+import '/constants/constants.dart';
+
+import '/repositories/genres_local_repository.dart';
+
+class MovieGenres extends StatelessWidget {
+  final List<int> movieGenres;
+
+  const MovieGenres({
+    super.key,
+    required this.movieGenres,
+  });
+
+  List<String> linkGenres(Map<int, String> genresMap) {
+    List<String> genresList = [];
+    for (int genreId in movieGenres) {
+      genresList.add(genresMap[genreId]!);
+    }
+    return genresList;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    IRepository genresRepository = GenresLocalRepository();
+
+    return FutureBuilder(
+      future: genresRepository.loadData(),
```

Comment:
We can add some Flutter best practices here. Please read this:
https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html

> If the future is created at the same time as the FutureBuilder, then every time the FutureBuilder's parent is rebuilt, the asynchronous task will be restarted. A general guideline is to assume that every build method could get called every frame, and to treat omitted calls as an optimization.


TLDR, you could make this widget Stateful, declare a Future<T> movies at the beginning, init that future in the initState method, and use that future in the FutureBuilder. Like here:
https://stackoverflow.com/a/68966640